### ISSUE TITLE
v0.2.2 build error

### ISSUE THREAD

**Issue by 0verse** [Neat-Lang/neat#25]
I'm trying to build v0.2.2-gcc and get this error:

```
ule_FF9AE93DC6BF49A7.o src/runtime.o
+ cat
+ '[' 64 == 32 ']'
+ ./neat_bootstrap -j src/main.nt -O -o neat
# stageM addQuasiQuotingMacro
$ gcc -c -fpic -fno-strict-aliasing .obj/module_A533D8106D4B4F61.c -o .obj/A533D8106D4B4F61.o [&28241]
./build.sh: line 56: 28230 Segmentation fault      ./neat_bootstrap -j src/main.nt -O -o neat
```

Edit: I get the same error with the llvm version:

```
+ ./neat_bootstrap -j -macro-backend=c src/main.nt -I/an/duyum/include -L-L/an/duyum/lib -O -version=LLVMBackend -o neat
# stageM addQuasiQuotingMacro
$ gcc -c -fpic -fno-strict-aliasing .obj/module_D4A1D9B6ECA5412D.c -o .obj/D4A1D9B6ECA5412D.o [&1852]
./build.sh: line 54:  1843 Segmentation fault      ./neat_bootstrap -j -macro-backend=c src/main.nt ${LLVM_NTFLAGS} -O -version=LLVMBackend -o neat
```

My system is based on musl libc, a custom x64 linux distro.

**Comment by FeepingCreature** [comment]
Uh. Good question, I have no idea what's happening there offhand. The bootstrap compiler seems to be segfaulting. Any chance you could get a backtrace of that command? (`./neat_bootstrap -j src/main.nt -O -o neat`)

**Comment by 0verse** [comment]
alpine 3.17 chroot gives same error:

```
+ cat
+ '[' 64 == 32 ']'
+ ./neat_bootstrap -j src/main.nt -O -o neat
./build.sh: line 56: 12832 Segmentation fault      ./neat_bootstrap -j src/main.nt -O -o neat
```

**Comment by 0verse** [comment]
alpine gdb gives more info:
```
(gdb) run -j src/main.nt -O -o neat
Starting program: /neat-v0.2.2-gcc/neat_bootstrap -j src/main.nt -O -o neat
[Detaching after vfork from child process 21205]
[New LWP 21206]
[New LWP 21207]
[New LWP 21208]
[New LWP 21209]
# stageM addQuasiQuotingMacro
[Detaching after fork from child process 21210]
$ gcc -c -fpic -fno-strict-aliasing .obj/module_A533D8106D4B4F61.c -o .obj/A533D8106D4B4F61.o [&21210]

Thread 4 "neat_bootstrap" received signal SIGSEGV, Segmentation fault.
[Switching to LWP 21208]
0x000055555587f020 in compiler130_module_neat_stuff_Resolver_resolve_either_void_compiler130_module_neat_base_class_Error_3_compiler130_module_neat_base_struct_Context_compiler130_module_neat_base_struct_LocRange_int ()
```

**Comment by FeepingCreature** [comment]
Well that's a completely different location for no reason...

Let me have a look.

**Comment by 0verse** [comment]
well now it gives this info:
```
(gdb) run -j src/main.nt -O -o neat
Starting program: /neat-v0.2.2-gcc/neat_bootstrap -j src/main.nt -O -o neat
[Detaching after vfork from child process 23613]
[New LWP 23614]
[New LWP 23615]
[New LWP 23616]
[New LWP 23617]

Thread 2 "neat_bootstrap" received signal SIGSEGV, Segmentation fault.
[Switching to LWP 23614]
0x00005555557e567e in compiler130_module_neat_stuff_parseExpressionLeaf_either_nullable_compiler130_module_neat_base_class_ASTSymbol_compiler130_module_neat_base_class_Error_2_compiler130_module_neat_base_class_Parser_compiler130_module_neat_base_class_LexicalContext ()
```

**Comment by FeepingCreature** [comment]
Ah. musl's stack size is too small for the compiler.

You can work around this by adding `-Wl,-z,stack-size=8388608` to the "gcc" line in `build.sh`.

How can I detect your system? I don't think I should always set that flag.

**Comment by 0verse** [comment]
ok the stack size fixed the segfault. You could check if "/lib/ld-musl-x86_64.so.1" is present in the system inside the build.sh file then you could add "-Wl,-z,stack-size=8388608" to the CFLAGS option.

**Comment by FeepingCreature** [comment]
Released 0.2.3 which should fix it without any changes to build scripts. Can you please confirm?

**Comment by 0verse** [comment]
Well it works fine with the new release, thanks for the time and effort!
